core/src/main/java/com/graphhopper/routing/ev/DecimalEncodedValueImpl.java

```diff
@@ -110,6 +113,8 @@ public void setDecimal(boolean reverse, IntsRef ref, double value) {
         if (Double.isNaN(value))
             throw new IllegalArgumentException("NaN value for " + getName() + " not allowed!");
 
+        realMax = Math.max(realMax, getNextStorableValue(value));
```

Comment:
> If we move it down we would have to support it for IntEncVal too.

Yes, but supporting it for IntEncVal is easier than supporting it for DecimalEncVal. It is the same except the matter is not complicated by defaultIsInfinity and useMaximumAsInfinity. Maybe we should think the other way around: Implement it for IntEncVal, then it will also work for DecimalEncVal by simply applying the factor. Well, at least as long as defaultIsInfinity and useMaximumAsInfinity are false, but I think it would be reasonable to simply not support getRealMax in this case.

I already tried this here: 304c05bf62786b3db56931e0a6915832b7cc8390, and to me it looks likes this is our best option currently.

Reply:
One thing that I actually find problematic is that the maximum could decrease again. If we set the speed for every edge once it is fine to use `realMax = Math.max(realMax, value)` here. But if the speed of some edges was set multiple time this does not work. For example if we set the speeds 10, 20 and 30 for edges 1,2 and 3 the maximum would be 30. If we then lower the speed of edge 3 to 10 the actual maximum would be 20, so even if we looked at the current value of edge 3 (30) and realized that it was the (in this case only) maximum we could not figure out that we need to lower the maximum to 20 (not 10) without looking at all the other edges (edge 2 specifically here).

Reply:
If we want the real maximum and have to expect that the values are set multiple times I think our only option is enforcing some kind of `baseGraph.freeze()` after which the graph and all the encoded values must not be changed again and looping over all edges once to find the maximum.

Reply:
> (Yes, or we calculate it independent from the EncodedValues)

After some discussion this does not seem to better. Determining the maximum could be better done outside of the encoded value after the graph has been finished, but storing and passing around the maximum value would be a lot more complicated also because only the weightings really know which max values they need to calculate the min weight (and only the weightings really know the weighting formula which is also required for this).